editor/plugins/polygon_2d_editor_plugin.cpp

```diff
@@ -612,6 +614,69 @@ void Polygon2DEditor::_uv_input(const Ref<InputEvent> &p_input) {
 					undo_redo->commit_action();
 				}
 
+				if (uv_move_current == UV_MODE_REMOVE_POINT) {
+					bool found_pt = false;
+
+					const real_t grab_threshold = (8 / uv_draw_zoom);
+
+					Vector2 closest_pt;
+					int closest_idx = 0;
+					real_t closest_dist = 1e10;
+
+					const int n_points = points_prev.size();
+					Vector2 p_pos = mtx.affine_inverse().xform(snap_point(Vector2(mb->get_position().x, mb->get_position().y)));
+
+					// Check if selected position is close to an existing point
+					for (int i = 0; i < n_points; i++) {
+						real_t d = points_prev[i].distance_to(p_pos);
+						if (d < closest_dist && d < grab_threshold) {
+							closest_dist = d;
+							closest_pt = points_prev[i];
+							closest_idx = i;
+							found_pt = true;
+						}
+					}
+
+					if (found_pt) { // Remove selected point
+						uv_create_uv_prev = node->get_uv();
+						uv_create_poly_prev = node->get_polygon();
+						uv_create_prev_internal_vertices = node->get_internal_vertex_count();
+						uv_create_colors_prev = node->get_vertex_colors();
+
+						undo_redo->create_action(TTR("Remove Point"));
+						undo_redo->add_do_method(node, "set_uv", node->get_uv());
```

Comment:
This part looks wrong. The `*_do_method` should have the new value, not the current one. Here it seems you have `uv_create_uv_prev == node->get_uv()`, so both the undo and redo values end up the same (Basically it means undoing then redoing will not work).

What should likely be done is move all the `undo_redo` calls after the actual polygon modification (line 676). You may also use `undo_redo->commit_action(false)` as the do action would already be done.